### DISCUSSION TITLE
LAB 07.01 Task3

### DISCUSSION THREAD

**Discussion by chape8** [rramosp/ai4eng.v1#209] [07 - DISEÑO DE ALGORITMOS]
Tengo problemas al utilizar minimize dado que no he podido saber bien como pasar los parámetros. Además, no veo como sacar del argumento params las variables X, t y b.
De esta manera lo he estado intentando
![lab7 01 error](https://user-images.githubusercontent.com/98787315/200014294-843feeba-76e5-4d9c-a04d-fa811cc52f56.png)
Entiendo que haciendo X= params[0] lo está interpretando como un flotante cuando debería ser una matriz, pero entonces no se como pasársela a la función loss si solo le ingresa params.

**Comment by imEag** [comment]
En este caso para usar el método [numpy.dot](https://numpy.org/doc/stable/reference/generated/numpy.dot.html) debes verificar que X sea de tipo array. Una forma de convertir un numpy.array64 a array es usar el método [numpy.asarrray](https://numpy.org/doc/stable/reference/generated/numpy.asarray.html). Sin embargo, analice si para multiplicar un float es necesario usar .dot o es mejor usar otro tipo de operación.

**Reply by chape8** [reply]
Es que esa multiplicación es una matricial (X y t asi como en la tarea 1), por eso utilicé el método dot, pero creo que esa manera como estoy accediendo a esos valores por medio de "params" no los está interpretando como matrices.

**Reply by imEag** [reply]
claro, cuando accedes a ellos de esa manera los valores obtenidos no son matrices, en este casi son nympy.float64
